### ISSUE TITLE
GUI Known Issues

### ISSUE THREAD

**Issue by MysticalOS** [DeadlyBossMods/DeadlyBossMods#204]
**High**
- [x] Option spacers now have following issues.
1. Are incorectly inserted (align incorrectly). 
2. Are not respected by next object in setpoint code and instead get ignored when next object sets point to previous option and not spacer.
3. Cause the area to have empty space at end, although this technically isn't a bug, just a side effect of number 2 since this space allotment should be used, when number 2 is fixed.

**Moderate**
- [x] SimpleHTML (option checkbox text) doesn't update wrapping at all on resize. It will update wrap on next UI load

**Minor**
- [x] SimpleHTML wrapping wraps on end of frame instead of end of frame area (Ie text can run under the scroll bar)
- [x] Load default options button doesn't reposition during or after drag. Wherever it appeared on load, is where it stays.
- [x] Currently GUI min size is capped at 800, 600 pixels to avoid some of the above issues from being bigger issues if user tries to make GUI too small. the cap will likely be lowered more when some of above stuff is fixed, but not made TOO much lower because there is no reason for GUI to be 30 by 30 pixels or anything goofy.
- [x] In rare cases it might think you grabbed resize corner when you haven't. I haven't been able to find exactly what causes this but if you accidentally resize window and actually drag corner off screen, there is an emergency reset size option in general options
- [x] boss mod Stats panels are auto sizing too much space, leaving a lot of empty space at bottom.
- [x] An EditBox needs to be added for both width and height to allow pixel perfectionists to set more specific size for GUI window.

**Classic Only**
- [ ] Dummy bars seem to misalign text with bar in GUI, fine elsewhere.

**Comment by Velinestra** [comment]
Message: Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:639: attempt to index global 'DBM_CORE_RANGECHECK_HEADER' (a nil value)
Time: Fri May 29 21:34:44 2020
Count: 2244
Stack: Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:639: attempt to index global 'DBM_CORE_RANGECHECK_HEADER' (a nil value)
Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:639: in function <Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:613>

Locals: activeRange = 10
restricted = true
tEnabled = true
rEnabled = false
reverse = nil
warnThreshold = 1
(*temporary) = <function> defined =[C]:-1
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = <function> defined =[C]:-1
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = <userdata>
(*temporary) = <userdata>
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index global 'DBM_CORE_RANGECHECK_HEADER' (a nil value)"
mainFrame = <unnamed> {
 0 = <userdata>
 hideTime = 0
 eventRegistered = true
 restrictions = true
 range = 10
 redCircleNumPlayers = 1
}
rangeCheck = <table> {
 Show = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:836
 IsRadarShown = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:906
 SetHideTime = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:914
 GetDistanceAll = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:922
 GetDistance = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:918
 UpdateRestrictions = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:910
 IsShown = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:902
 Hide = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:882
}
textFrame = DBMRangeCheck {
 0 = <userdata>
 text = <unnamed> {
 }
 lines = <table> {
 }
 inRangeText = <unnamed> {
 }
}
radarFrame = DBMRangeCheckRadar {
 circle = <unnamed> {
 }
 inRangeText = <unnamed> {
 }
 dots = <table> {
 }
 text = <unnamed> {
 }
 0 = <userdata>
 background = <unnamed> {
 }
}
prevRange = 0
prevThreshold = 0
pixelsperyard = 0
GetBestMapForUnit = <function> defined =[C]:-1
rotation = 0
pi2 = 6.283185
GetPlayerFacing = <function> defined =[C]:-1
sin = <function> defined =[C]:-1
cos = <function> defined =[C]:-1
GetNumGroupMembers = <function> defined =[C]:-1
unitList = <table> {
 1 = "raid1"
 2 = "raid2"
 3 = "raid3"
 4 = "raid4"
 5 = "raid5"
 6 = "raid6"
 7 = "raid7"
 8 = "raid8"
 9 = "raid9"
 10 = "raid10"
 11 = "raid11"
 12 = "raid12"
 13 = "raid13"
 14 = "raid14"
 15 = "raid15"
 16 = "raid16"
 17 = "raid17"
 18 = "raid18"
 19 = "raid19"
 20 = "raid20"
}
UnitExists = <function> defined =[C]:-1
UnitIsUnit = <function> defined =[C]:-1
UnitIsDeadOrGhost = <function> defined =[C]:-1
UnitIsConnected = <function> defined =[C]:-1
UnitInPhase = <function> defined =[C]:-1
itsBCAgain = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:77
UnitName = <function> defined =[C]:-1
RAID_CLASS_COLORS = <table> {
 HUNTER = <table> {
 }
 WARRIOR = <table> {
 }
 PALADIN = <table> {
 }
 MAGE = <table> {
 }
 PRIEST = <table> {
 }
 SHAMAN = <table> {
 }
 WARLOCK = <table> {
 }
 DEMONHUNTER = <table> {
 }
 DEATHKNIGHT = <table> {
 }
 DRUID = <table> {
 }
 MONK = <table> {
 }
 ROGUE = <table> {
 }
}
setDot = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:572
updateSound = <function> defined @Interface\AddOns\DBM-Core\DBM-RangeCheck.lua:386
prevNumClosePlayer = 0
prevclosestRange = 0
prevType = 0
circleColor = 0
prevColor = 0

**Comment by Velinestra** [comment]
Message: Interface\AddOns\DBM-Core\DBM-Core.lua:4444: bad argument #1 to 'mrandom' (interval is empty)
Time: Fri May 29 18:59:34 2020
Count: 1
Stack: Interface\AddOns\DBM-Core\DBM-Core.lua:4444: bad argument #1 to 'mrandom' (interval is empty)
[C]: ?
Interface\AddOns\DBM-Core\DBM-Core.lua:4444: in function <Interface\AddOns\DBM-Core\DBM-Core.lua:4432>
(tail call): ?
Interface\AddOns\DBM-Core\DBM-Core.lua:4976: in function `handler'
Interface\AddOns\DBM-Core\DBM-Core.lua:843: in function <Interface\AddOns\DBM-Core\DBM-Core.lua:830>

Locals: (*temporary) = 1
(*temporary) = 0

**Comment by cwwood0322** [comment]
When logging into a toon I get "WARNING: An old version of DBM-SpellTimers is present that breaks DBM. DBM-SpellTimers must be updated or removed for DBM to function properly. Update will require a game restart." I looked there is not a DBM module that I got off of twitch that is called DBM-Spell Timers. This started last night and continued today even after doing the latest update to DBM. In the addon folder in WOW it is the third one down on the list under Deadly Boss Mods Core.
